torch_geometric/data/temporal.py

```diff
@@ -97,14 +98,74 @@ def __init__(
         super().__init__()
         self.__dict__['_store'] = GlobalStorage(_parent=self)
 
+        if t is None:
+            t = torch.full_like(src, 0)
+        else:
+            src, dst, t, msg = sort_events(src, dst, t, msg)
         self.src = src
         self.dst = dst
         self.t = t
         self.msg = msg
 
+        self._unique_time, inverse_indices, counts = torch.unique_consecutive(
+            t, return_counts=True, return_inverse=True)
+        self._slices = cumsum(counts)
+
         for key, value in kwargs.items():
             setattr(self, key, value)
 
+    def snapshot(self, start: Optional[int] = None, end: Optional[int] = None):
+        """Returns the graph snapshot where events start from
+        :obj:`start`-th and end at :obj:`end`-th time step.
+        """
+        start = start or 0
+        end = end or -1
+        slices = self._slices
+        temporal_data = self[slices[start]:slices[end]]
+        edge_index = torch.stack([temporal_data.src, temporal_data.dst], dim=0)
+        edge_attr = getattr(temporal_data, 'msg', None)
+        time_attr = temporal_data.t
+
+        return Data(edge_index=edge_index, edge_attr=edge_attr,
+                    time_attr=time_attr)
+
+    def update(self, src, dst: Optional[Tensor] = None,
+               t: Optional[Tensor] = None, msg: Optional[Tensor] = None):
+        """Updates the temporal graph with newly incoming events
+        :obj:`(src, dst, t, msg)`
+        """
+        src = torch.tensor(src).view(-1)
+
+        last_time = self._unique_time[-1]
+
+        if t is None:
+            t = torch.full_like(src, last_time + 1)
+        else:
+            t = torch.tensor(t).view(-1)
```

Comment:
Why do we take the view? Can we assume that users will provide a certain input shape?